File: scripts/process-lifecycle/process.pl
#!/usr/bin/env perl

use v5.38;

sub worker {
    say "\tChild process (PID: $$) started.";
    sleep(2);
    say "\tChild process (PID: $$) finished.";
    exit(0);
}

sub parent {
    say "Parent process (PID: $$) started.";
    my $pid = fork();
    if ($pid == 0) {
        worker();
    } else {
        say "\tChild process (PID: $pid) created.";
        waitpid($pid, 0);
        say "Parent process (PID: $$) finished.";
    }
}

parent();

File: scripts/process-lifecycle/process.py
#!/usr/bin/env python3

import os
import time

def worker():
    print(f"\tChild process (PID: {os.getpid()}) started.")
    time.sleep(2)
    print(f"\tChild process (PID: {os.getpid()}) finished.")
    os._exit(0)

def parent():
    print(f"Parent process (PID: {os.getpid()}) started.")
    pid = os.fork()
    if pid == 0:
        worker()
    else:
        print(f"\tChild process (PID: {pid}) created.")
        os.waitpid(pid, 0)
        print(f"Parent process (PID: {os.getpid()}) finished.")

if __name__ == "__main__":
    parent()
